### PR TITLE
migration for kraken.services

### PR DESCRIPTION
**Pull request by DStorck**

### CODE REVIEW

File: ansible/roles/kraken.config/files/config.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ version: v1alpha
 definitions:
   helmConfigs:
     - &defaultHelm
-      name: defaultHelm
+      name: helm
```
Comment (davidewatson) [inline ansible/roles/kraken.config/files/config.yaml]: Unlike the defaultEtcd to etcd name change, which was only required if you used a master config file with the k2recon branch, the helm name should probably stay defaultHelm.

File: ansible/roles/kraken.config/tasks/main.yaml [hunk 2]
```diff
@@ -25,6 +25,17 @@
     msg: "Make sure that the 'cluster' is defined in {{ config_file }}"
   when: kraken_config.cluster | is_empty
 
+- name: Warn if still using old style configuration format
```
Comment (davidewatson) [inline ansible/roles/kraken.config/tasks/main.yaml]: I am not sure why this change, and others, are appearing in your PR, but we should try to clean them up so that the commit only includes the changes for #274.

File: ansible/roles/kraken.services/tasks/kill-services.yaml [hunk 3]
```diff
@@ -1,4 +1,32 @@
...
+    cluster_namespaces: "{{ cluster_namespaces | unique | reject('search', 'kube-system') | list }}"
+
+- name: See if tiller rc if present
+  shell: >
+    kubectl --kubeconfig={{ kubeconfig }} get deployment {{ tiller }} --namespace=kube-system
+  register: tiller_present
+  when: kubeconfig | is_file
+  ignore_errors: yes
+
+- name: Create Helm home
```
Comment (davidewatson) [inline ansible/roles/kraken.services/tasks/kill-services.yaml]: I'm curious under what conditions this directory wouldn't exist. It doesn't hurt anything, but if it's useful we should add a comment as to why. Otherwise we should remove it.

File: ansible/roles/kraken.services/tasks/run-services.yaml [hunk 4]
```diff
@@ -1,4 +1,24 @@
...
+    kubeconfig: "{{ config_base | expanduser }}/{{ cluster.name }}/admin.kubeconfig"
+    helm_home: "{{ config_base | expanduser }}/{{ cluster.name }}/.helm"
+    cluster_namespaces: []
+- set_fact:
+    cluster_namespaces: "{{ cluster_namespaces + [item.namespace | default('kube-system')] }}"
+  with_items: "{{ cluster_services }}"
+- set_fact:
+    cluster_namespaces: "{{ cluster_namespaces | unique | reject('search', 'kube-system') | list }}"
+
+- name: Create Helm home
```
Comment (davidewatson) [inline ansible/roles/kraken.services/tasks/run-services.yaml]: Since you do this above as well, maybe we should only do it once in main.yaml?

Comment (DStorck) [inline ansible/roles/kraken.services/tasks/run-services.yaml]: are you talking about creating the path for helm_home in main.yaml?
I wanted to do that but I don't have access to the helm_home variable

File: ansible/roles/kraken.config/files/config.yaml (hunk 1 above)
Comment (DStorck) [inline ansible/roles/kraken.config/files/config.yaml]: no problem

File: ansible/roles/kraken.services/tasks/kill-services.yaml (hunk 3 above)
Comment (DStorck) [inline ansible/roles/kraken.services/tasks/kill-services.yaml]: I can do whatever makes more sense but I don't have the context to make that determination.

File: ansible/roles/kraken.services/tasks/run-services.yaml (hunk 4 above)
Comment (davidewatson) [inline ansible/roles/kraken.services/tasks/run-services.yaml]: Ah, I see. The helm_home and kubeconfig variables are really just conveniences. If you remove them from the defaults, you should be able to iterate over the tasks in main ```with_items: "{{ kraken_config.clusters }}"``` using their definitions.

Comment (DStorck) [inline ansible/roles/kraken.services/tasks/run-services.yaml]: I can put `create helm home` back in main.yaml if i do something like this: ```- name: Create Helm home
  file: >
    path= "{{ config_base | expanduser }}/{{ kraken_config.cluster }}/.helm" 
    state=directory
``` but i thought we were trying to change `kraken_config.cluster` to `cluster.name` wherever we could for the new config style
